### PR TITLE
Refactor to use dwolla/jenkins-agent-core:debian upstream; update nvm to 0.34.0

### PR DESCRIPTION
**Pull request by bpholt**
This project predates `dwolla/jenkins-agent-core:debian`, but now that it's available, this refactoring removes the duplicated logic.

I also noticed that a new version of nvm was available, so I upgraded it as well.

### CODE REVIEW

**Comment by bpholt** [comment]
I can't figure out how to get the ci/dockercloud integration working on this repo. @coreyjonoliver do you have any ideas?

File: Dockerfile [hunk 1]
```diff
@@ -1,61 +1,20 @@
...
-WORKDIR ${JENKINS_HOME}
-COPY jenkins-agent install.sh verify.sh /usr/local/bin/
+COPY install.sh verify.sh /usr/local/bin/
 
-# apt-key loop inspired by https://github.com/nodejs/docker-node/issues/340#issuecomment-321669029
+USER root
 RUN set -ex && \
     apt-get update && \
     apt-get install -y \
-        apt-transport-https \
```
Comment (nickolashkraus) [inline Dockerfile:19]: I am guessing these packages are installed on the `jenkins-agent-core:debian` image?

**Review by nickolashkraus — APPROVED** [review]
Looks good to me! 👍

File: Dockerfile (hunk 1 above)
Comment (bpholt) [reply Dockerfile:19]: Yep! 

https://github.com/Dwolla/jenkins-agent-docker-core/blob/79af34db214e0c1f58be199bbceb29a374e94ed0/debian/Dockerfile#L18-L28

**Comment by bpholt** [comment]
Re: https://github.com/Dwolla/jenkins-agent-docker-nvm/pull/21#issuecomment-470296895, I ended up changing the config on the Docker Cloud side to only tag things on the `master` branch, made Autotests was set to Internal and External Pull Requests, and then force pushed a new commit to trigger the builds.

**Pull request merged**